spec/requests/applicants/create_address_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'POST /v1/applicants/:applicant_id/addresses', type: :request do
+  let(:application) { create(:application, :with_applicant) }
+  let(:applicant) { application.applicant }
+  let(:application_id) { application.application_ref }
+  let(:applicant_id) { applicant.id }
+  let(:params) do
+    {
+      address_line_one: '123',
+      address_line_two: 'High Street',
+      city: 'London',
+      county: 'Greater London',
+      postcode: 'SW1H9AJ'
+    }
+  end
+  let(:post_request) do
+    -> { post "/v1/applicants/#{applicant_id}/addresses", params: params.to_json, headers: json_headers }
+  end
```

Comment:
I think this is over complicating things. If you change it to:
```ruby
let(:post_request) do
  post "/v1/applicants/#{applicant_id}/addresses", params: params.to_json, headers: json_headers
end
```
It can be called with `post_request`, rather than `post_request.call`

Reply:
This is more to express explicitly that we're actually performing the request, rather than defining it, I don't mind either way.